worlds/tunic/__init__.py

```diff
@@ -332,8 +371,75 @@ def remove_filler(amount: int) -> None:
             if tunic_item.name in slot_data_item_names:
                 self.slot_data_items.append(tunic_item)
 
+        # pull out the filler so that we can place it manually during pre_fill
+        self.grass_fill = []
+        self.grass_fill_locations = []
+        self.non_grass_fill = []
+        self.non_grass_fill_locations = []
+        # grass and non-grass get their own pools
+        if self.options.grass_randomizer:
+            filler = self.grass_fill
+            fill_locations = self.grass_fill_locations
+        else:
+            filler = self.non_grass_fill
+            fill_locations = self.non_grass_fill_locations
+        if self.options.local_fill > 0 and self.multiworld.players > 1:
+            # skip items marked local or non-local, let fill deal with them in its own way
+            # discard grass from non_local if it's meant to be limited
+            if self.settings.limit_grass_rando:
+                self.options.non_local_items.value.discard("Grass")
+            all_filler: List[TunicItem] = []
+            non_filler: List[TunicItem] = []
+            for tunic_item in tunic_items:
+                if (tunic_item.classification in [ItemClassification.filler, ItemClassification.trap]
+                        and tunic_item.name not in self.options.local_items
+                        and tunic_item.name not in self.options.non_local_items):
+                    all_filler.append(tunic_item)
+                else:
+                    non_filler.append(tunic_item)
+            amount_to_local_fill = int(self.options.local_fill.value * len(all_filler) / 100)
+            # we need to reserve a couple locations so that we don't fill up every sphere 1 location
+            reserved_locations: Set[str] = set(self.random.sample(sphere_one, 2))
+            viable_locations = [loc for loc in self.multiworld.get_unfilled_locations(self.player)
+                                if loc.name not in reserved_locations
+                                or loc.progress_type is not LocationProgressType.EXCLUDED
+                                or loc.name in self.options.priority_locations.value]
+            amount_to_local_fill = min(amount_to_local_fill, len(viable_locations))
+            self.random.shuffle(all_filler)
+            filler += all_filler[:amount_to_local_fill]
+            del all_filler[:amount_to_local_fill]
+            self.random.shuffle(viable_locations)
+            fill_locations += viable_locations[:amount_to_local_fill]
```

Comment:
Hm.

The previous implementation meant that if any of the viable locations got sth plandoed onto it, it was fine.
Now, `fill_locations` is exactly the same size as `filler` for each world, meaning that if *any* of those locations get plandoed, one of the filler items is not placed.

I think it was better when you saved all viable locations until stage_pre_fill, to keep your options open.

Reply:
I <3 plando